### PR TITLE
cleanup the repo

### PR DESCRIPTION
**Pull request by mangalaman93**
The PR does the following:
  * improves README
  * removes deepsource.yml and tesh.sh
  * fixes import order
  * adds golangci.yml and fixes linter issues
  * updates copyright year to 2023
  * fixes Makefile for proto generation

### CODE REVIEW

**Comment by joshua-goldstein** [comment]
Can we add a .go-version file?  It will be needed for workflows:
https://github.com/dgraph-io/dgraph/blob/main/.go-version

**Comment by mangalaman93** [comment]
> Can we add a .go-version file? It will be needed for workflows: https://github.com/dgraph-io/dgraph/blob/main/.go-version

Done

File: acl_test.go [hunk 1]
```diff
@@ -37,8 +37,7 @@ var (
 	unusedgroup  = "unused"
 	devgroup     = "dev"
 
-	dgraphAddress = "127.0.0.1:9180"
```
Comment (joshua-goldstein) [inline acl_test.go:40]: we don't use this anymore?

File: examples_test.go [hunk 2]
```diff
@@ -677,22 +678,22 @@ func ExampleTxn_Mutate_facets() {
 		NameOrigin: "Indonesia",
 		DType:      []string{"Person"},
 		Friends: []Person{
-			Person{
+			{
```
Comment (joshua-goldstein) [inline examples_test.go:681]: this is implied (inherited) from the slice type?

File: protos/Makefile [hunk 3]
```diff
@@ -1,23 +1,54 @@
...
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+GOGO_PATH   := github.com/gogo/protobuf
 
-PROTO_PATH := ${GOPATH}/src:.
-PROTO_PATH := ${PROTO_PATH}:${GOPATH}/src/github.com/gogo/protobuf
+TMPDIR     := $(shell mktemp -d)
+PROTO_PATH := ${TMPDIR}/src:.
```
Comment (joshua-goldstein) [inline protos/Makefile:20]: what is the diff between this protos Makefile and the one in the dgraph repo?  can we keep them the same?

**Comment by joshua-goldstein** [comment]
Looks good, just had some questions about the protos makefile.  We should keep them similar in structure.

File: examples_test.go (hunk 2 above)
Comment (mangalaman93) [reply examples_test.go:681]: Yes

File: protos/Makefile (hunk 3 above)
Comment (mangalaman93) [reply protos/Makefile:20]: They can't be exactly the same because the proto there depends on this proto.

File: acl_test.go (hunk 1 above)
Comment (mangalaman93) [reply acl_test.go:40]: I guess. I could use it though, let me check

Comment (mangalaman93) [reply acl_test.go:40]: not using this, but added a constant for the address

File: protos/Makefile [hunk 4]
```diff
@@ -1,23 +1,54 @@
...
+copy-protos:
+	@mkdir -p ${TMPDIR}/src/${GOGO_PATH}/gogoproto
+	@cp $(shell go list -m -f "{{.Dir}}" ${GOGO_PATH})/gogoproto/gogo.proto \
+	 ${TMPDIR}/src/${GOGO_PATH}/gogoproto/gogo.proto
+
+.PHONY: regenerate
+regenerate: tidy-deps copy-protos check clean
 	@protoc \
+		--proto_path=/usr/local/include \
+		--proto_path=/usr/include \
```
Comment (joshua-goldstein) [inline protos/Makefile:50]: are these really necessary?  if yes, then shouldn't we have a check to make sure `/usr/include/google/protobuf` or `/usr/local/include/google/protobuf` are present?  We do this in the dgraph repo:  https://github.com/dgraph-io/dgraph/blob/07d3765280bb87231194ef0acc905c35cd8ab2a6/protos/depcheck.sh#L46

File: protos/Makefile [hunk 5]
```diff
@@ -1,23 +1,54 @@
...
+check:
+	echo "Installing proto libraries to versions in go.mod." ; \
+		go install github.com/golang/protobuf/protoc-gen-go ; \
+		go install github.com/gogo/protobuf/protoc-gen-gogofaster
+
+.PHONY: copy-protos
+copy-protos:
+	@mkdir -p ${TMPDIR}/src/${GOGO_PATH}/gogoproto
+	@cp $(shell go list -m -f "{{.Dir}}" ${GOGO_PATH})/gogoproto/gogo.proto \
+	 ${TMPDIR}/src/${GOGO_PATH}/gogoproto/gogo.proto
```
Comment (joshua-goldstein) [inline protos/Makefile:44]: This copy might not be necessary; in dgraph repo we need to copy three different proto files into a common directory, but here we only have one proto file.  Am I misunderstanding something?

File: protos/Makefile (hunk 4 above)
Comment (mangalaman93) [reply protos/Makefile:50]: Done

File: protos/Makefile (hunk 5 above)
Comment (mangalaman93) [reply protos/Makefile:44]: Done

**Pull request merged**